store/store.go

```diff
@@ -1806,7 +1807,8 @@ func (s *Store) isStaleRead(freshness int64) bool {
 	if freshness == 0 || s.raft.State() == raft.Leader {
 		return false
 	}
-	return time.Since(s.raft.LastContact()).Nanoseconds() > freshness
+	return time.Since(s.raft.LastContact()).Nanoseconds() > freshness ||
+		s.raft.CommitIndex() != s.DBAppliedIndex()
```

Comment:
Having `s.raft.CommitIndex() != s.DBAppliedIndex()` makes the condition too strict, especially during a large update that can still finish within the allowed staleness.

Also, I noticed `s.DBAppliedIndex()` is protected by a mutex. What do you think about instead using a https://pkg.go.dev/sync/atomic#Uint64?